### PR TITLE
A few IBC fixes

### PR DESCRIPTION
**Pull request by vmarkushin**
- Fix `UpdateClient` event
- Fix `PacketData` and `PrefixedDenom` serialization
- Fix transfer `Acknowledgement` serialization

### CODE REVIEW

File: ibc/modules/src/applications/transfer/acknowledgement.rs [hunk 1]
```diff
@@ -13,66 +13,51 @@
...
 
 impl Acknowledgement {
 	pub fn success() -> Self {
-		Self::Success(ACK_SUCCESS_B64.to_vec())
+		Self { result: Some(ACK_SUCCESS_B64.to_string()), error: None }
 	}
 
 	pub fn from_error(err: Error) -> Self {
-		Self::Error(format!("{}: {}", ACK_ERR_STR, err))
+		Self { result: None, error: Some(err.to_string()) }
```
Comment (blasrodri) [inline ibc/modules/src/applications/transfer/acknowledgement.rs:40]: By changing this aren’t we going to have problems with other implementations? An enum is different to a struct. 
Just worried about introducing a new protocol that nobody else other than pallet ibc would understand.

File: ibc/modules/src/applications/transfer/acknowledgement.rs [hunk 2]
```diff
@@ -13,66 +13,51 @@
...
-	fn as_ref(&self) -> &[u8] {
-		match self {
-			Acknowledgement::Success(b) => b.as_slice(),
-			Acknowledgement::Error(s) => s.as_bytes(),
+	pub fn into_result(self) -> Result<String, String> {
+		match (self.result, self.error) {
+			(Some(r), None) if r == ACK_SUCCESS_B64 => Ok(r),
+			(Some(r), None) => Err(r),
+			(None, Some(e)) => Err(e),
+			_ => unreachable!("invalid response"),
```
Comment (blasrodri) [inline ibc/modules/src/applications/transfer/acknowledgement.rs:52]: Do we want to panic in the runtime? @Wizdave97

**Review by Wizdave97 — APPROVED** [review]

File: ibc/modules/src/applications/transfer/acknowledgement.rs (hunk 1 above)
Comment (Wizdave97) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:40]: They're  using a struct now in ibc-go

File: ibc/modules/src/applications/transfer/acknowledgement.rs (hunk 2 above)
Comment (Wizdave97) [inline ibc/modules/src/applications/transfer/acknowledgement.rs:52]:
```suggestion
			_ => Err("invalid ack".to_string()),
```

**Review by Wizdave97 — CHANGES_REQUESTED** [review]

File: ibc/modules/src/applications/transfer/acknowledgement.rs (hunk 1 above)
Comment (Wizdave97) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:40]: https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#acknowledgement-envelope
[type Acknowledgement struct {](https://github.com/cosmos/ibc-go/blob/b2fb1192245134607e0e7a294d09d2aa1145f017/modules/core/04-channel/types/channel.pb.go#L402)

Comment (blasrodri) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:40]: True. Go has no enums. I wanted to confirm that we weren’t introducing a conflict

Comment (vmarkushin) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:40]: I can make it an enum, but then serialization will be implemented manually (not through `derive`)

Comment (vmarkushin) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:40]: Actually, no, I can make it an enum

**Review by blasrodri — DISMISSED** [review]

File: ibc/modules/src/applications/transfer/acknowledgement.rs [hunk 3]
```diff
@@ -13,66 +13,49 @@
...
+use serde::{Deserialize, Serialize};
 
 /// A string constant included in error acknowledgements.
 /// NOTE: Changing this const is state machine breaking as acknowledgements are written into state
 pub const ACK_ERR_STR: &str = "error handling packet on destination chain: see events for details";
-pub const ACK_SUCCESS_B64: &[u8] = b"AQ==";
+pub const ACK_SUCCESS_B64: &str = "AQ==";
 
-#[derive(Clone, Debug)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
```
Comment (Wizdave97) [inline ibc/modules/src/applications/transfer/acknowledgement.rs:26]: @vmarkushin Does this still serialize into a json object?

Comment (vmarkushin) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:26]: Yes, I've checked

Comment (vmarkushin) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:26]: The variant name is the key in JSON:
`Ack::Result` -> `{ "result": "..." }`
`Ack::Error` -> `{ "error": "..." }`

Comment (Wizdave97) [reply ibc/modules/src/applications/transfer/acknowledgement.rs:26]: Okay cool

**Review by Wizdave97 — DISMISSED** [review]

**Review by blasrodri — APPROVED** [review]

**Pull request merged**